src/async.c

```diff
@@ -43,21 +113,83 @@ static int luv_new_async(lua_State* L) {
     return luv_error(L, ret);
   }
   data = luv_setup_handle(L, ctx);
-  data->extra = (luv_thread_arg_t*)malloc(sizeof(luv_thread_arg_t));
+  luv_async_arg_t* asarg = (luv_async_arg_t*)malloc(sizeof(luv_async_arg_t));
+  memset(asarg, 0, sizeof(luv_async_arg_t));
+  asarg->max = max;
+  ret = uv_mutex_init(&asarg->mutex);
+  if (ret < 0) { // unlikely
+    abort();
```

Comment:
we shouldn't be aborting the program, this should be reported as a fail

Reply:
Fixed in commit 9fa42ef602913926b2fef0752b5d0ee8325f96e8